src/DistributedLock.Core/Internal/Data/DatabaseCommand.cs

```diff
@@ -153,9 +153,7 @@ public IDbDataParameter AddParameter(string? name = null, object? value = null,
         Invariant.Require(cancellationToken.CanBeCanceled);
 
         using var _ = await this.AcquireConnectionLockIfNeeded(isConnectionMonitoringQuery).ConfigureAwait(false);
-        // Note: for now we cannot pass cancellationToken to PrepareAsync() because this will break on Postgres which
-        // is the only db we currently support that needs Prepare currently. See https://github.com/npgsql/npgsql/issues/4209
-        await this.PrepareIfNeededAsync(CancellationToken.None).ConfigureAwait(false);
+        await this.PrepareIfNeededAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This is the only code change. I didn't add any comments because this is the expected behavior and the original behavior was an exception.